apollo-ios/Sources/Apollo/RequestContext.swift

```diff
@@ -10,3 +10,10 @@ import ApolloAPI
 /// This allows the various interceptors to make modifications, or perform actions, with information
 /// that they cannot get just from the existing operation. It can be anything that conforms to this protocol.
 public protocol RequestContext {}
+
+/// A specialized request context that specifies configuration details for the URLRequest.
+public protocol RequestConfigurationContext: RequestContext {
```

Comment:
I think this should be a specific protocol for this feature named something like `RequestContextTimeoutConfigurable`. Any additional features we want to add onto the `RequestContext` in the future should be their own individual protocols.

That way people can opt-in to the configuration options they want and not be forced to implement configuration for other options they don't need to get the ones they do. This is also more likely to prevent us from needing breaking changes in the future when we want to make changes to these things (or at least limit the surface area effected by changes).